ci/matrix.yaml

```diff
@@ -54,7 +60,12 @@ workflows:
     - {jobs: ['build'], project: 'cudax', ctk: ['curr'], std: 'all', cxx: ['gcc', 'clang'], cpu: 'arm64'}
     - {jobs: ['test'],  project: 'cudax', ctk: ['curr'], std: 20,    cxx: ['gcc', 'clang', 'msvc'], gpu: 'rtx2080'}
     # Python and c/parallel jobs:
-    - {jobs: ['test'], project: ['cccl_c_parallel', 'python'], gpu: 'rtx2080'}
+    - {jobs: ['build_cuda_cccl', 'build_cuda_cooperative', 'build_cuda_parallel'], project: 'python', py_version: '3.10'}
+    - {jobs: ['build_cuda_cccl', 'build_cuda_cooperative', 'build_cuda_parallel'], project: 'python', py_version: '3.12'}
+    - {jobs: ['build_cuda_cccl', 'build_cuda_cooperative', 'build_cuda_parallel'], project: 'python', py_version: '3.13'}
+    - {jobs: ['test_cuda_cccl', 'test_cuda_cooperative', 'test_cuda_parallel'], project: 'python', gpu: 'rtxa6000', py_version: '3.10'}
+    - {jobs: ['test_cuda_cccl', 'test_cuda_cooperative', 'test_cuda_parallel'], project: 'python', gpu: 'rtxa6000', py_version: '3.12'}
+    - {jobs: ['test_cuda_cccl', 'test_cuda_cooperative', 'test_cuda_parallel'], project: 'python', gpu: 'rtxa6000', py_version: '3.13'}
```

Comment:
This entire block can be replaced with a single line:

```suggestion
    - {jobs: ['test'], project: 'python', py_version: ['3.10', '3.12', '3.13'], gpu: 'rtxa6000`}
```

1. The py_version list will be expanded
2. The `test` job will automatically add the required `build` job
3. The `test` and `build` jobs will be replaced according to the entries in the `python` job map.

We haven't tested having multiple producers like this before, let me know if it doesn't work as expected.

Reply:
Nice - it looks like this is working as expected.

Reply:
I see that the PR matrix was updated, but not the override. Make sure we test this with the override updated to make sure that the artifact passing is handled correctly.

Reply:
Ah, right. Trying that now.

Reply:
Nice, looks like it worked as expected 🥳

Reply:
It worked! https://github.com/NVIDIA/cccl/pull/4679/#issuecomment-2881340084